contracts/extensions/VotingReputation.sol

```diff
@@ -0,0 +1,890 @@
+  )
+    public
+  {
+    uint256 domainSkillId = colony.getDomain(_domainId).skillId;
+    uint256 actionDomainSkillId = getActionDomainSkillId(_action);
+
+    if (domainSkillId != actionDomainSkillId) {
+      uint256 childSkillId = colonyNetwork.getChildSkillId(domainSkillId, _childSkillIndex);
+      require(childSkillId == actionDomainSkillId, "voting-rep-invalid-domain-id");
+    }
+
+    createPoll(_target, _action, _domainId, domainSkillId, _key, _value, _branchMask, _siblings);
+  }
+
+  /// @notice Stake on a poll
+  /// @param _pollId The id of the poll
+  /// @param _permissionDomainId The domain where the extension has the arbitration permission
+  /// @param _childSkillIndex For the domain in which the poll is occurring
+  /// @param _vote The side being supported (0 = NAY, 1 = YAY)
+  /// @param _amount The amount of tokens being staked
+  /// @param _key Reputation tree key for the staker/domain
+  /// @param _value Reputation tree value for the staker/domain
+  /// @param _branchMask The branchmask of the proof
+  /// @param _siblings The siblings of the proof
+  function stakePoll(
+    uint256 _pollId,
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    uint256 _vote,
+    uint256 _amount,
+    bytes memory _key,
+    bytes memory _value,
+    uint256 _branchMask,
+    bytes32[] memory _siblings
+  )
+    public
+  {
+    Poll storage poll = polls[_pollId];
+    require(_vote <= 1, "voting-rep-bad-vote");
+    require(getPollState(_pollId) == PollState.Staking, "voting-rep-staking-closed");
+
+    uint256 requiredStake = getRequiredStake(_pollId);
+    uint256 amount = min(_amount, sub(requiredStake, poll.stakes[_vote]));
+    uint256 stakerTotalAmount = add(stakes[_pollId][msg.sender][_vote], amount);
+
+    require(
+      stakerTotalAmount <= getReputationFromProof(_pollId, msg.sender, _key, _value, _branchMask, _siblings),
+      "voting-rep-insufficient-rep"
+    );
+    require(
+      stakerTotalAmount >= wmul(requiredStake, userMinStakeFraction),
+      "voting-rep-insufficient-stake"
+    );
+
+    colony.obligateStake(msg.sender, poll.domainId, amount);
+    colony.transferStake(_permissionDomainId, _childSkillIndex, address(this), msg.sender, poll.domainId, amount, address(this));
+
+    // Update the stake
+    poll.stakes[_vote] = add(poll.stakes[_vote], amount);
+    stakes[_pollId][msg.sender][_vote] = stakerTotalAmount;
+
+    // Increment counter & extend claim delay if staking for an expenditure state change
+    if (poll.stakes[YAY] == requiredStake && _vote == YAY && getSig(poll.action) == CHANGE_FUNCTION) {
+      bytes32 structHash = hashExpenditureStruct(poll.action);
+      expenditurePollCounts[structHash] = add(expenditurePollCounts[structHash], 1);
+      bytes memory claimDelayAction = createClaimDelayAction(poll.action, UINT256_MAX);
+      executeCall(_pollId, claimDelayAction);
+    }
+
+    // Move to second staking window once one side is fully staked
```

Comment:
What is the difference between the first and the second staking windows? We're tracking `poll.events[STAKE1_END]`, but we never use it?

Reply:
First window is for either side to stake, second window is for the second side to stake. If one side stakes, the other has 72 hours to stake, else the motion fails.

Reply:
Okay, that makes sense, but we're not using the `STAKE1_END` timestamp anywhere, so why are we tracking it? Why do we not just have `STAKE_END`, and if one side fully stakes it (and subsequent timestamps) get bumped by `stakingPeriod`, which is essentially is what it happening with the `STAKE2_END` timestamp being tracked.

Reply:
Great point!